Don't send queries to llm

johnswanson opened the pull request:
This introduces `envelope/llm-message`, a private function that formats
messages for the LLM. Currently it does two things: removes `query`s and
json-encodes the structured content.

There's also `envelope/llm-history`, which replaces
`envelope/full-history` by providing the full history appropriate for
the LLM (instead of the full "internal" history for our own use).

johnswanson committed:
Don't send queries to the LLM

This introduces `envelope/llm-message`, a private function that formats
messages for the LLM. Currently it does two things: removes `query`s and
json-encodes the structured content.

There's also `envelope/llm-history`, which replaces
`envelope/full-history` by providing the full history appropriate for
the LLM (instead of the full "internal" history for our own use).

metamben reviewed (commented):
I like the "unified" tool call organization and I prefer the second option. I think the `llm-history` (in fact, something similar for the whole `llm-request`) will be needed anyway.

metamben commented on enterprise/backend/src/metabase_enterprise/metabot_v3/envelope.clj:
```diff
@@ -41,6 +40,34 @@
+  "Formats a message for the LLM. Removes things we don't want the LLM to see (e.g. `query`) and stringifies structured
+  content."
+  [{:as msg :keys [structured-content content]}]
+  (cond-> msg
+    (and
+     (is-tool-call-response? msg)
+     (is-query? structured-content)) (update :structured-content dissoc :query)
+    structured-content (as-> msg*
+                             (assoc msg* :content (or content
+                                                      (json/encode (:structured-content msg*)))))
+    true (dissoc :structured-content)))
```
I found this difficult to parse. Maybe lining up the updating parts can help.

johnswanson committed:
Rework confusing fn

Split this out into a few functions:

- `stringified-content` takes a message and returns the stringified
content (from `structured-content` or `content` as appropriate)

- `stringify-content` takes a message and returns a message with a
stringified `:content` form (and no `:structured-content`)

- `maybe-remove-query` takes a message and removes the `:query` if
necessary

Now `llm-message` just threads the message through `maybe-remove-query`
and then `stringify-content` rather than a confusing `cond->` form.
